app/Model/ObjectTemplate.php

```diff
@@ -316,4 +316,41 @@ public function setActive($id)
         }
         return 1;
     }
+
+    public function getRawFromDisk($uuidOrName)
+    {
+        $templates = $this->readTemplatesFromDisk();
+        if (Validation::uuid($uuidOrName)) {
+            return $templates[$uuidOrName];
+        } else {
+            foreach ($templates as $uuid => $template) {
+                if ($template['name'] == $uuidOrName) {
```

Comment:
@Rafiot  suggests that multiple templates might have the same name.  This would randomly bring back only one of them.  Please revisit.

Reply:
Exactly! You can't have duplicated name on the filesystem, so there is no need to return more than once. Same goes for UUIDs, you should not have 2 objects with the same UUID.

Reply:
The method for checking the name uses the string from inside the file, not the directory name.  Where is the code that reads these in to the database and/or UI, and what does it do?  I recommend extending this method with a comparison of the directory name with the name inside the file, and log an error/skip the entry if they do not match.